lib/stdlib/src/gen.erl

```diff
@@ -546,3 +548,41 @@ format_status_header(TagLine, RegName) when is_atom(RegName) ->
     lists:concat([TagLine, " ", RegName]);
 format_status_header(TagLine, Name) ->
     {TagLine, Name}.
+
+-spec format_status(Mod :: module(), Opt :: terminate | normal, Status, Args) ->
+          ReturnStatus when
+      Status :: #{ atom() => term() },
+      ReturnStatus :: #{ atom() => term(), '$status' => term()},
+      Args :: list(term()) | undefined.
+format_status(Mod, Opt, Status, Args) ->
+    case {erlang:function_exported(Mod, format_status, 1),
+          erlang:function_exported(Mod, format_status, 2)} of
+        {true, _} ->
+            try Mod:format_status(Status) of
+                NewStatus when is_map(NewStatus) ->
+                    MergedStatus = maps:merge(Status, NewStatus),
+                    case maps:size(MergedStatus) == maps:size(NewStatus) of
+                        true ->
+                            MergedStatus;
+                        false ->
+                            ?LOG_ERROR("~p:format_status(~p) returned an invalid map",
+                                       [Mod, Status]),
+                            Status#{ 'EXIT' => atom_to_list(Mod) ++ ":format_status/1 returned an invalid map" }
+                    end;
+                _ ->
+                    Status#{ 'EXIT' => atom_to_list(Mod) ++ ":format_status/1 did not return a map" }
+            catch
+                E:R:ST ->
+                    ?LOG_ERROR("Calling ~p:format_status/1 failed: ~tp:~tp~n~tp",
```

Comment:
We should remove this ?LOG_ERROR in order to not accidentally leak any state. The same with the invalid map log

Reply:
fixed.